response/max_age.go

```diff
@@ -2,12 +2,41 @@ package response
 
 import (
 	"context"
+	"regexp"
+	"strings"
 
+	"github.com/ONSdigital/dp-legacy-cache-proxy/config"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
-func maxAge(ctx context.Context, path string) int {
-	log.Info(ctx, "Calculating max-age for "+path)
-	// TODO To be implemented as part of DIS-411
-	return 42
+func maxAge(ctx context.Context, uri string, cfg *config.Config) int {
+	log.Info(ctx, "Calculating max-age for "+uri)
+
+	if isVersionedURI(uri) || isOnsURI(uri) || isLegacyAssetURI(uri) {
+		return int(cfg.CacheTimeLong.Seconds())
+	}
+
+	return int(cfg.CacheTimeDefault.Seconds())
+}
+
+func isVersionedURI(uri string) bool {
+	matched, _ := regexp.MatchString(`/previous/v[0-9]+`, uri)
```

Comment:
```suggestion
	matched, _ := re.Match(uri)
```
would argue that ignoring this error is worrying

perhaps a future PR could compile the global regex at startup time

```
func init() {
  re = regexp.MustCompile(`/previous/v[0-9]+`)
}
```